### DISCUSSION TITLE
Does /dl endpoint requires to be logged in?

### DISCUSSION THREAD

**Discussion by Geobert** [moriturus/ktra#12] [Q&A] [answered]
Hi!

After successfully tested Ktra on my computer, we're trying to deploy it in our infra and when trying to fetch a crate in our private index, we get this log:

```
Feb 16 10:30:15.523  WARN request{method=GET path=/dl/our_crate/0.1.0/download version=HTTP/1.1 remote.addr=10.0.9.5:45020}: warp::filters::trace: unable to serve request (client error) status=400 error=Rejection([MethodNotAllowed, MethodNotAllowed, MethodNotAllowed, MethodNotAllowed, MethodNotAllowed, MethodNotAllowed, MethodNotAllowed, MethodNotAllowed, MissingHeader { name: "Authorization" }])
```
Why `/dl` need an `Authorization` header?

To be fair, the exact situation is this:
I tested Ktra on my computer and publish `our_crate`.
Now, I'm hitting the deployed Ktra, which has not receive the publish request but uses the private index with `our_crate` in it. I don't know if this can be an issue.

**Comment by moriturus** [comment, marked as answer]
Hi @Geobert, I'm sorry for my late response. 

So, `/dl` API does not need `Authorization` header. The log message is not helpful to debug for now. This may come from `warp`'s implementation.

Can you download your crate file with a http client like `curl`?  
I'm not familiar with Windows, but a sample command is here:

```
curl -i YOUR_HOST_OR_IP:PORT/dl/YOUR_CRATE_NAME/VERSION/download
```

and it seems succeed in downloading if you get below message.

```
HTTP/1.1 200 OK
content-length: 808
content-type: application/octet-stream
accept-ranges: bytes
last-modified: Tue, 16 Feb 2021 11:23:18 GMT
date: Tue, 16 Feb 2021 20:24:02 GMT
```

**Reply by Geobert** [reply]
No worries for the "late" response, it's a side project after all :D 

I'll try that after solving the checksum issue, because I've wipe my index and I can't publish with the checksum error :)
